eqs/src/eth_polling.rs

```diff
@@ -114,44 +116,76 @@ impl EthPolling {
             state_persistence,
             last_event_index,
             next_block_to_query,
+            max_blocks_to_query: opt.max_ether_blocks(),
             pending_commit_event: Vec::new(),
             connection,
             num_confirmations: opt.num_confirmations,
         }
     }
 
     pub async fn check(&mut self) -> Result<u64, async_std::io::Error> {
-        let latest_block_number = self
-            .connection
-            .provider
-            .get_block_number()
-            .await
-            .expect("Could not fetch latest block number")
-            .as_u64();
-
-        let fetch_until = match self
-            .num_confirmations
-            .latest_confirmed_block_number(latest_block_number)
-        {
-            Some(n) => n,
-            // There is no confirmed block to fetch yet.
-            None => return Ok(0),
-        };
+        loop {
+            let fetch_from = self.next_block_to_query;
+            let latest_block_number = self
+                .connection
+                .provider
+                .get_block_number()
+                .await
+                .expect("Could not fetch latest block number")
+                .as_u64();
+
+            let fetch_latest = match self
+                .num_confirmations
+                .latest_confirmed_block_number(latest_block_number)
+            {
+                Some(n) => n,
+                // There is no confirmed block to fetch yet.
+                None => return Ok(0),
+            };
+
+            if fetch_latest < fetch_from {
+                break;
+            }
+
+            let fetch_until = if fetch_latest > fetch_from + self.max_blocks_to_query {
```

Comment:
@nyospe Is this `min(fetch_latest, fetch_from + self.max_blocks_to_query)`?

Reply:
I tried using min initially, but something went wrong. I don't remember what, but I must've changed whatever caused it, because it seems to work now.